isomorphic-fetch/index.d.ts

```diff
@@ -3,182 +3,107 @@
 // Definitions by: Todd Lucas <https://github.com/toddlucas>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-type RequestType = "" | "audio" | "font" | "image" | "script" | "style" |
-    "track" | "video";
-type RequestDestination = "" | "document" | "embed" | "font" | "image" |
-    "manifest" | "media" | "object" | "report" | "script" | "serviceworker" |
-    "sharedworker" | "style" | "worker" | "xslt";
-type RequestMode = "navigate" | "same-origin" | "no-cors" | "cors";
-type RequestCredentials = "omit" | "same-origin" | "include";
-type RequestCache =
-    "default" | "no-store" | "reload" | "no-cache" | "force-cache" |
-    "only-if-cached";
-type RequestRedirect = "follow" | "error" | "manual";
-
-type ResponseType = "basic" | "cors" | "default" | "error" | "opaque" |
-    "opaqueredirect";
-
-type ReferrerPolicy = "" | "no-referrer" | "no-referrer-when-downgrade" |
-    "same-origin" | "origin" | "strict-origin" | "origin-when-cross-origin" |
-    "strict-origin-when-cross-origin" | "unsafe-url";
-
-interface HeadersInterface {
-    append(name: string, value: string): void;
-    delete(name: string): void;
-    get(name: string): string | null;
-    getAll(name: string): string[];
-    has(name: string): boolean;
-    set(name: string, value: string): void;
-
-    // TODO: iterable<string, string>;
-    forEach(callback: (value: string, index: number, headers: HeadersInterface) => void, thisArg?: any): void;
-    // NOTE: The following are supported by whatwg-fetch but not node-fetch.
-    // entries(): IterableIterator<[string, string]>;
-    // keys(): IterableIterator<string>;
-    // values(): IterableIterator<string>;
+interface ForEachCallback {
+    (keyId: any, status: string): void;
 }
 
-type HeadersInit = Headers | string[] | { [index: string]: string };
-
-declare class Headers implements HeadersInterface {
-    constructor(init?: HeadersInit);
+interface Headers {
     append(name: string, value: string): void;
     delete(name: string): void;
+    forEach(callback: ForEachCallback): void;
     get(name: string): string | null;
-    getAll(name: string): string[];
     has(name: string): boolean;
     set(name: string, value: string): void;
-
-    forEach(callback: (value: string, index: number, headers: HeadersInterface) => void, thisArg?: any): void;
 }
 
-interface BodyInterface {
-    bodyUsed: boolean;
-    arrayBuffer(): Promise<ArrayBuffer>;
-    blob(): Promise<Blob>;
-    formData(): Promise<FormData>;
-    json(): Promise<any>;
-    json<T>(): Promise<T>;
-    text(): Promise<string>;
+declare var Headers: {
+    prototype: Headers;
+    new(init?: any): Headers;
 }
 
-declare class Body implements BodyInterface {
-    bodyUsed: boolean;
+interface Body {
+    readonly bodyUsed: boolean;
     arrayBuffer(): Promise<ArrayBuffer>;
     blob(): Promise<Blob>;
```

Comment:
This makes the lib unusable without `"dom"` lib. Although that was the case before.